inc/class-thumbnails.php

```diff
@@ -92,7 +93,16 @@ public function check_orphaned( $post_id, $post ) {
 				if ( strpos( $meta_key, '_' . $provider . '_' ) !== false && strpos( $meta_key, '_url' ) === false ) {
 					$id = str_replace( 'embed_privacy_thumbnail_' . $provider . '_', '', $meta_key );
 					
-					if ( strpos( $post->post_content, $id ) === false ) {
+					$missing_id = strpos( $post->post_content, $id ) === false;
+					$missing_url = true;
+					if ( $missing_id && isset( $metadata[ $meta_key . '_url' ] ) ) {
+						$url = $metadata[ $meta_key . '_url' ];
+						if ( is_array( $url ) ) {
+							$url = reset( $url );
+						}
+						$missing_url = strpos( $post->post_content, $url ) === false;
+					}
+					if ( $missing_id && $missing_url ) {
 						if ( ! $this->is_in_use( $meta_value, $post_id, $global_metadata ) ) {
```

Comment:
Please combine this and the if statement from above into one

Reply:
I am not sure about this one. The first if is used to check if `missing_url` shall be defined. It is defined only if `missing_id` is false. If the ID is already missing, there is no point in also checking the URL (which means a second expensive `strpos` operation).

How would you combine them?

Reply:
Sorry, I had edited the previous comment a couple of times because I mixed up some boolean things. 😄

Reply:
I’m not sure if we talk about the same. I mean these two:
```
if ( $missing_id && $missing_url ) {
	if ( ! $this->is_in_use( $meta_value, $post_id, $global_metadata ) ) {
```

Just combine them to:
```
if ( $missing_id && $missing_url && ! $this->is_in_use( $meta_value, $post_id, $global_metadata ) ) {
```

I don’t see a reason why these need to be processed separately.

Reply:
Now I get what you are saying. That's true, I will merge them.